Components/Installer/PaymentMethods/PaymentMethodsInstaller.php

```diff
@@ -574,27 +570,60 @@ private function setPaymentActive(Payment $payment, $isActive)
     /**
      * Returns a collection of active payment methods from the Mollie API.
      *
-     * @return null|BaseCollection|MethodCollection
+     * @return array
      */
     private function getActivePaymentMethodsFromMollie()
     {
-        /** @var MethodCollection $methods */
-        $methods = null;
-
-        try {
-            $methods = $this->mollieApiClient->methods->allActive(
-                [
-                    'resource' => 'orders',
-                    'includeWallets' => 'applepay',
-                ]
-            );
-        } catch (ApiException $e) {
-            $this->logger->error(
-                'Error when loading active payment methods from Mollie',
-                [
-                    'error' => $e->getMessage(),
-                ]
-            );
+        $methods = [];
+        $mollieApiFactory = new MollieApiFactory(clone $this->config, $this->logger);
+        $shopService = new ShopService($this->modelManager);
+        $shops = $shopService->getAllShops();
+
+        if (empty($shops)) {
+            return $methods;
+        }
+
+        foreach($shops as $shop) {
+            try {
+                $mollieApiClient = $mollieApiFactory->create($shop->getId());
+            } catch (ApiException $e) {
+                $this->logger->error(
+                    sprintf('Error when creating a Mollie API client for shop %s', $shop->getName()),
+                    [
+                        'error' => $e->getMessage(),
+                    ]
+                );
+
+                continue;
```

Comment:
could you change to just 1 try catch within the foreach please
its a all or nothing, but still fault tolerant on shop level, thats a good one

Reply:
Will do!